packages/backend-core/src/middleware/passport/datasource/google.js

```diff
@@ -0,0 +1,76 @@
+const { getScopedConfig } = require("../../../db/utils")
+const { getGlobalDB } = require("../../../tenancy")
+const google = require("../google")
+const { Configs, Cookies } = require("../../../constants")
+const { clearCookie, getCookie } = require("../../../utils")
+const { getDB } = require("../../../db")
+
+async function preAuth(passport, ctx, next) {
+  const db = getGlobalDB()
+  // get the relevant config
+  const config = await getScopedConfig(db, {
+    type: Configs.GOOGLE,
+    workspace: ctx.query.workspace,
+  })
+  const publicConfig = await getScopedConfig(db, {
+    type: Configs.SETTINGS,
+  })
+  let callbackUrl = `${publicConfig.platformUrl}/api/global/auth/datasource/google/callback`
```

Comment:
Just to confirm we are ok for tenancy here? 
In the login callbacks we add the tenantid to be able to fetch the correct global db on the way back in. It's safe to say that this will only ever be hit while authenticated right, as in when the datasource is being configured?

Reply:
Yep, the user will always be logged into the platform already before this happens, so our middleware should take care of it - if I remember correctly